### PR TITLE
Improve dl3006

### LINKED ISSUE #161: DL3006 rule fails in multi-stage build FROM
- [x] This is a bug report

### Expected behavior
The rule should not fail when `FROM` base image uses an other stage in a multi-stage build Dockerfile

### Actual behavior
The rule fails

### Steps to reproduce the behavior

The second build stage `unit-test` of the following Dockerfile is using the previous stage as it's base image which is totally fine but the rule require that we specify a tag and no tag exists for a stage.

The Dockerfile is an example and the problem with the rule is not related to the language at all.

Using `hadolint/hadolint:latest` image to lint a multi-stage Dockerfile

```Dockerfile
FROM golang:1.9.3-alpine3.7 AS build

... (building my go app here)

FROM build as unit-test

... (testing my go app here)

FROM alpine:3.7
...
```

### PR DESCRIPTION
**Pull request by lorenzo**
Fixes #161

### CODE REVIEW

File: package.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: hadolint
-version: '1.2.6'
+version: '1.3.0'
```
Comment (lorenzo) [inline package.yaml]: Forgot to push this to master

File: src/Hadolint/Rules.hs [hunk 2]
```diff
@@ -211,14 +211,20 @@ noAptGetUpgrade = instructionRule code severity message check
...
     code = "DL3006"
     severity = WarningC
     message = "Always tag the version of an image explicitly."
-    check (From (UntaggedImage image _)) = image == "scratch"
+    check (From (UntaggedImage image _)) = image == "scratch" || image `elem` aliasedImages
     check (From TaggedImage {}) = True
     check _ = True
+    aliasedImages =
+        unImageAlias <$>
+        [a | From (UntaggedImage _ (Just a)) <- instr] ++
```
Comment (lukasmartinelli) [inline src/Hadolint/Rules.hs]: I don't quite understand what this does anymore :( My haskell is decaying.

Comment (lorenzo) [reply src/Hadolint/Rules.hs]: It gets all elements of the instr list that can be pattern matched to that expression

Comment (lorenzo) [reply src/Hadolint/Rules.hs]: and returns the variable a, as the elements of the resulting list

**Comment by zemanlx** [comment]
Nice work. Thanks. I will probably add check for lint and version in Travis later.

**Pull request merged**